### PR TITLE
33 extend data model

### PR DESCRIPTION
**Pull request by wenwenchenbosch**
solve #33

### CODE REVIEW

**Comment by wenwenchenbosch** [comment]
using `updateMetaData <path> <json/file>` in testclient to update a datamodel

**Review by SebastianSchildt — CHANGES_REQUESTED** [review]
JWT access permission need to be reeavluated when modifying tree. Reproduce

- Use token with *=rw acces
- Extend tree
- Try to acces newly added item
  -  Expectation: Works
  - Situation: 403
- Authorize with same token again
- Access newly added item: Works now

File: certificates/jwt/super-admin-metadata.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "sub": "kuksa.val",
+  "iss": "Eclipse KUKSA Dev",
+  "admin": true,
+  "metadata": "rw",
```
Comment (SebastianSchildt) [inline certificates/jwt/super-admin-metadata.json:5]: modifyTree? modifyVSSTree?

**Comment by wenwenchenbosch** [comment]
> JWT access permission need to be reeavluated when modifying tree. Reproduce
> 
>     * Use token with *=rw acces
> 
>     * Extend tree
> 
>     * Try to acces newly added item
>       
>       * Expectation: Works
>       * Situation: 403
> 
>     * Authorize with same token again
> 
>     * Access newly added item: Works now

this bug fixed now

File: docker/Dockerfile-Jenkins-Build-Env [hunk 2]
```diff
@@ -4,7 +4,7 @@ FROM ubuntu:18.04
 
 # Install all the pre-requisits we need in the docker image
 RUN apt-get update \
-  && apt-get install -y qemu-user-static docker.io sudo bzip2
+  && apt-get install -y qemu-user-static docker.io sudo bzip2 ccache
```
Comment (SebastianSchildt) [inline docker/Dockerfile-Jenkins-Build-Env:7]: Isn't ccache useless here? We never run a compiler in this container

Comment (wenwenchenbosch) [reply docker/Dockerfile-Jenkins-Build-Env:7]: with `docker build`: `RUN make -j8` the local gcc should be used. But using ccache is still not configured within jenkins. There are still more works to do. It is a preparation.

File: certificates/jwt/super-admin-metadata.json (hunk 1 above)
Comment (wenwenchenbosch) [reply certificates/jwt/super-admin-metadata.json:5]: really necessary to change?

Comment (SebastianSchildt) [reply certificates/jwt/super-admin-metadata.json:5]: I think `modifytree`  describes more clearly what we are doing here. But now I am puzzled about "r" .. are we also protecting getMetadata now?

Comment (wenwenchenbosch) [reply certificates/jwt/super-admin-metadata.json:5]: no, default always read-only. But it is also possible to extend it

Comment (wenwenchenbosch) [reply certificates/jwt/super-admin-metadata.json]: I update the key work to `modifyTree`. Question: do we need two token?

**Review by SebastianSchildt — APPROVED** [review]
Working fine, can be merged now @rohoet

File: docker/Dockerfile-Jenkins-Build-Env (hunk 2 above)
Comment (rohoet) [reply docker/Dockerfile-Jenkins-Build-Env:7]: Is ccache now required here or not? Its license is not compliant with EPL.

Comment (SebastianSchildt) [reply docker/Dockerfile-Jenkins-Build-Env:7]: Hi. 
It is just our build environment, it has nothing to do with the software  (and it is a complete ubuntu 18.04, we can not clear everything in there)

Comment (wenwenchenbosch) [reply docker/Dockerfile-Jenkins-Build-Env]: `ccache` removed

**Pull request merged**